webapp/tests/karma/unit/services/lineage-model-generator.js

```diff
@@ -157,6 +157,24 @@ describe('LineageModelGenerator service', () => {
       });
     });
 
+    it('should merge lineage with undefined members', () => {
+      const contact = { _id: 'a', name: '1', parent: { _id: 'b', parent: { _id: 'c', parent: { _id: 'd' } } } };
+      const parent = { _id: 'b', name: '2', parent: { _id: 'c', parent: { _id: 'd' } } };
+      dbQuery.resolves({ rows: [{ doc: contact, key: ['a', 0] }, { doc: parent,  key: ['a', 1] }, { key: ['a', 2] }, { key: ['a', 3] }] });
+      const expected = {
+        _id: 'a',
+        doc: {
+          _id: 'a',
+          name: '1',
+          parent: { _id: 'b', name: '2', parent: { _id: 'c', parent: { _id: 'd' } } }
+        },
+        lineage: [{ _id: 'b', name: '2', parent: { _id: 'c', parent: { _id: 'd' } } }, undefined, undefined]
```

Comment:
I'm concerned about the case where we have a lineage like `[a, b, undefined, undefined, e]`, if higher ancestors are present again after having some which are absent. Would you add that case to the test or add a second test for it?

Reply:
I added a test for that case, but I wouldn't be concerned much about that one. Missing a parent along the way would be a major bug, no matter how you cut it :)